src/marqo/client.py

```diff
@@ -112,21 +114,25 @@ def create_index(
             image_preprocessing_method=image_preprocessing_method,
             settings_dict=settings_dict, inference_node_type=inference_node_type, storage_node_type=storage_node_type,
             storage_node_count=storage_node_count, replicas_count=replicas_count,
-            inference_node_count=inference_node_count,
+            inference_node_count=inference_node_count, wait_for_readiness=wait_for_readiness
         )
 
-    def delete_index(self, index_name: str) -> Dict[str, Any]:
+    def delete_index(self, index_name: str, wait_for_readiness=True) -> Dict[str, Any]:
         """Deletes an index
 
         Args:
             index_name: name of the index
-
+            wait_for_readiness: Marqo Cloud specific, whether to wait until
+                operation is completed or to proceed without waiting for status,
+                won't do anything if config.is_marqo_cloud=False
         Returns:
             response body about the result of the delete request
         """
         try:
             res = self.http.delete(path=f"indexes/{index_name}")
-            cloud_wait_for_index_status(self.http, index_name, enums.IndexStatus.DELETED)
+            if self.config.is_marqo_cloud and wait_for_readiness:
+                cloud_wait_for_index_status(self.http, index_name, enums.IndexStatus.DELETED)
+                return res
```

Comment:
No return for non-cloud? Or indentation is wrong?

Reply:
actioned